### PR TITLE
Add FN_NO_RETURN macro.

### PR DESCRIPTION
**Pull request by shkshk90**
Change all instances of `__attribute__((__noreturn__))` to a macro in [`meson.build`](meson.build) / [`build.auto.h.in`](src/build.auto.h.in).

As compiler attributes written in the form of `__attribute__` is not supported by _msvc_, this is one of several pull requests to make the code-base more robust, and allows using _msvc_-style attributes later.

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -155,6 +155,10 @@ endif
 # Set configuration path
 configuration.set_quoted('CFGOPTDEF_CONFIG_PATH', get_option('configdir'), description: 'Configuration path')
 
+# Set FN_NO_RETURN macro
+configuration.set('FN_NO_RETURN', '__attribute__((__noreturn__))', description: 'No return function attribute')
+
+
```
Comment (dwsteele) [inline meson.build:161]: Remove the extra linefeed here. We never have more than one blank line for spacing.

File: src/common/error.h [hunk 2]
```diff
@@ -314,28 +314,28 @@ jmp_buf *errorInternalJump(void);
...
     const ErrorType *errorType, const char *fileName, const char *functionName, int fileLine, const char *format, ...)
-    __attribute__((format(printf, 5, 6))) __attribute__((__noreturn__));
+    __attribute__((format(printf, 5, 6)));
 
 // Throw a system error
-void errorInternalThrowSys(
+FN_NO_RETURN void errorInternalThrowSys(
     int errNo, const ErrorType *errorType, const char *fileName, const char *functionName, int fileLine, const char *message)
-    __attribute__((__noreturn__));
+   ;
```
Comment (dwsteele) [inline src/common/error.h:333]: This semicolon should be at the end of the line above.

**Review by dwsteele — CHANGES_REQUESTED** [review]
Looks pretty good overall but see comments.

**Review by dwsteele — APPROVED** [review]
Looks good, thanks!

In the future keep in mind that it is up to the reviewer to resolve comments, i.e. when the reviewer feels that the issue has been addressed. We just add a comment with the commit where the issue was fixed, then the reviewer checks and resolves.

File: meson.build [hunk 3]
```diff
@@ -155,6 +155,9 @@ endif
 # Set configuration path
 configuration.set_quoted('CFGOPTDEF_CONFIG_PATH', get_option('configdir'), description: 'Configuration path')
 
+# Set FN_NO_RETURN macro
+configuration.set('FN_NO_RETURN', '__attribute__((__noreturn__))', description: 'Indicate that a function does not return')
```
Comment (eli-schwartz) [inline meson.build:159]: Note for the future, you can do configure time checks to see if this is supported via:

https://mesonbuild.com/Reference-manual_returned_compiler.html#compilerhas_function_attribute

Comment (shkshk90) [reply meson.build:159]: Thanks for this tip.

I went a bit through meson's reference, but didn't catch it.
I will definitely make use of that, when I start implementing the Windows port.

Thank you :).

**Pull request merged**